diff --git a/CMakeLists.txt b/CMakeLists.txt
index 23ae1dd63..1bc8e7166 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,8 @@ project(dpctl
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
 
 # Option to generate code coverage report using llvm-cov and lcov.
 option(DPCTL_GENERATE_COVERAGE
diff --git a/examples/pybind11/onemkl_gemv/CMakeLists.txt b/examples/pybind11/onemkl_gemv/CMakeLists.txt
index b9c4b087e..7e2bfb0ff 100644
--- a/examples/pybind11/onemkl_gemv/CMakeLists.txt
+++ b/examples/pybind11/onemkl_gemv/CMakeLists.txt
@@ -30,6 +30,7 @@ find_library(mkl_sycl NAMES mkl_sycl PATHS ${MKL_LIBRARY_DIR} REQUIRED)
 find_library(mkl_intel_ilp64 NAMES mkl_intel_ilp64 PATHS ${MKL_LIBRARY_DIR} REQUIRED)
 find_library(mkl_tbb_thread NAMES mkl_tbb_thread PATHS ${MKL_LIBRARY_DIR} REQUIRED)
 find_library(tbb NAMES tbb PATHS ${TBB_LIBRARY_DIR} REQUIRED)
+find_library(OpenCL NAMES OpenCL REQUIRED)
 
 set(py_module_name _onemkl)
 
@@ -64,7 +65,7 @@ target_include_directories(standalone_cpp
   PUBLIC ${MKL_INCLUDE_DIR} ${TBB_INCLUDE_DIR} sycl_gemm
   )
 target_link_libraries(standalone_cpp
-    PRIVATE ${mkl_sycl} ${mkl_intel_ilp64} ${mkl_tbb_thread} ${mkl_core} ${tbb}
+    PRIVATE ${mkl_sycl} ${mkl_intel_ilp64} ${mkl_tbb_thread} ${mkl_core} ${tbb} ${OpenCL}
 )
 
 set(ignoreMe "${SKBUILD}")
diff --git a/libsyclinterface/CMakeLists.txt b/libsyclinterface/CMakeLists.txt
index a3ceca466..4c6987901 100644
--- a/libsyclinterface/CMakeLists.txt
+++ b/libsyclinterface/CMakeLists.txt
@@ -67,8 +67,10 @@ if(DPCTL_ENABLE_L0_PROGRAM_CREATION)
             OUTPUT_VARIABLE LIBZE_LOADER_FILENAME
             OUTPUT_STRIP_TRAILING_WHITESPACE
             ERROR_STRIP_TRAILING_WHITESPACE
-            )
+        )
+        set(LIBZE_LOADER_FILENAME ${LIBZE_LOADER_FILENAME} PARENT_SCOPE)
     endif()
+    set(LIBZE_LOADER_FILENAME ${LIBZE_LOADER_FILENAME} PARENT_SCOPE)
 endif()
 
 if (UNIX)
@@ -86,8 +88,12 @@ if (UNIX)
     OUTPUT_STRIP_TRAILING_WHITESPACE
     ERROR_STRIP_TRAILING_WHITESPACE
   )
+  set(LIBCL_LOADER_FILENAME ${LIBCL_LOADER_FILENAME} PARENT_SCOPE)
 endif()
 
+message(STATUS "LIB_ZE: ${LIBZE_LOADER_FILENAME}")
+message(STATUS "LIB_CL: ${LIBCL_LOADER_FILENAME}")
+
 configure_file(
     ${CMAKE_CURRENT_SOURCE_DIR}/include/Config/dpctl_config.h.in
     ${CMAKE_CURRENT_SOURCE_DIR}/include/Config/dpctl_config.h
diff --git a/libsyclinterface/include/Config/dpctl_config.h.in b/libsyclinterface/include/Config/dpctl_config.h.in
index d630060b9..44f5c53ec 100644
--- a/libsyclinterface/include/Config/dpctl_config.h.in
+++ b/libsyclinterface/include/Config/dpctl_config.h.in
@@ -27,7 +27,7 @@
 
 /* Defined when dpctl was built with level zero program creation enabled. */
 #cmakedefine DPCTL_ENABLE_L0_PROGRAM_CREATION                                  \
-    @DPCTL_ENABLE_L0_PROGRAM_CREATION @
+    @DPCTL_ENABLE_L0_PROGRAM_CREATION@
 
 /* The DPCPP version used to build dpctl */
 #define DPCTL_DPCPP_VERSION "@IntelSycl_VERSION@"
